tests: tests/subsys/logging/log_backend_fs : NXP S32 series build failure

**Describe the bug**

all S32 platform build fails

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - "mr_canhubk3",
   "s32z2xxdc2/s32z270/rtu0",
    "s32z2xxdc2@D/s32z270/rtu0",
    "s32z2xxdc2/s32z270/rtu1",
    "s32z2xxdc2@D/s32z270/rtu1",
 - What have you tried to diagnose or workaround this issue?
 - git bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?

```
- dc97cc0cd529702508e1ff0943dfee3e1f7fa7fb is the first bad commit
commit dc97cc0cd529702508e1ff0943dfee3e1f7fa7fb
Author: Joakim Andersson <joerchan@gmail.com>
Date:   Tue Feb 4 09:53:53 2025 +0100

    tests: logging: Convert unit test to include the full FS backend
    
    Convert testing of the logging backend for file-system to include the
    full implementation.
    Need to disable logging and just include the unit under test to make
    this simpler.
    The API functions for the backend can then be called on the backend
    object.
    
    Signed-off-by: Joakim Andersson <joerchan@gmail.com>

 subsys/logging/backends/log_backend_fs.c           |  5 ----
 tests/subsys/logging/log_backend_fs/CMakeLists.txt | 15 ++++++++++-
 tests/subsys/logging/log_backend_fs/Kconfig        |  8 ------
 tests/subsys/logging/log_backend_fs/prj.conf       |  6 -----
 .../logging/log_backend_fs/src/log_fs_test.c       | 30 +++++++++++++++++++++-
 5 files changed, 43 insertions(+), 21 deletions(-)
 delete mode 100644 tests/subsys/logging/log_backend_fs/Kconfig
```

**To Reproduce**

```
west twister -p ucans32k1sic/s32k146  -T tests/subsys/logging/log_backend_fs
```

**Expected behavior**

build pass

**Impact**

zephyr enablement for S32 series

**Logs and console output**

```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/subsys/logging/log_backend_fs/src/log_fs_test.c:35:17: error: unknown type name 'log_format_func_t' 35 | FAKE_VALUE_FUNC(log_format_func_t, log_format_func_t_get, uint32_t); | ^~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/testsuite/include/zephyr/fff.h:8632:19: note: in definition of macro 'EXPAND' 8632 | #define EXPAND(x) x | ^ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/testsuite/include/zephyr/fff.h:8658:29: note: in expansion of macro 'EXPAND' 8658 | #define FUNC_VALUE_(N, ...) EXPAND(FUNC_VALUE_N(N, __VA_ARGS__)) | ^~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/testsuite/include/zephyr/fff.h:8660:30: note: in expansion of macro 'EXPAND' 8660 | #define FUNC_VALUE_N(N, ...) EXPAND(FAKE_VALUE_FUNC##N(__VA_ARGS__)) | ^~~~~~ 
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v4.0.0-4796-ge4043824b9b3

--- a/tests/subsys/logging/log_backend_fs/src/log_fs_test.c
+++ b/tests/subsys/logging/log_backend_fs/src/log_fs_test.c
@@ -16,6 +16,7 @@
 #include <zephyr/ztest.h>
 #include <zephyr/fs/fs.h>
 #include <zephyr/fff.h>
+#include <zephyr/logging/log_backend.h>
 
 #define DT_DRV_COMPAT zephyr_fstab_littlefs
 #define TEST_AUTOMOUNT DT_PROP(DT_DRV_INST(0), automount)

